storage/plugins/mobile-api/src/Controllers/CatalogController.php

```
@@ -1035,6 +1102,22 @@ private function sortSpec(?string $sort): array
                 return ['key' => 'title_desc', 'col' => 'l.titolo', 'dir' => 'DESC',
                         'order' => 'l.titolo DESC, l.id DESC', 'field' => 'titolo'];
+            case 'author_asc':
+            case 'author_desc':
+                // Sort by the primary-author display name (issue #282). The
+                // sort value is the SAME correlated subquery that populates the
+                // `autore` result column, so it matches what the app shows.
+                // COALESCE(..., '') keeps books with no author sorting as an
+                // empty string on BOTH sides of the keyset comparison — a raw
+                // NULL sort column would make `NULL {cmp} ?` false and silently
+                // drop authorless books from every page after the first.
+                $authorSortSql = \App\Support\AuthorName::displaySql('a');
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Support\AuthorName' in method 'sortSpec'. (undefined)

(StaticAccess)